educe/graph.py

```diff
@@ -502,16 +502,15 @@ def cdu_members(self, cdu, deep=False):
         that are members of (members of ..) this CDU.
         """
 
+        hyperedge = self.edgeform(cdu)
+        members = set(self.links(hyperedge))
+
         if deep:
-            members = set()
-            for m in self.cdu_members(cdu):
-                members.add(m)
+            for m in list(members):
                 if self.is_cdu(m):
-                    members.update(self.cdu_members(m,deep))
-            return frozenset(members)
-        else:
-            hyperedge = self.edgeform(cdu)
-            return frozenset(self.links(hyperedge))
+                    members.update(self.cdu_members(m, True))
```

Comment:
Makes absolutely explicit that the value of `deep` is carried from the current state to recursive calls. Mostly a hunch, though.